### PR TITLE
London-9 - Hadi A.P - Full-Stack-Project-Assessment

### PR DESCRIPTION
**Pull request by hadipournigjeh**

### CODE REVIEW

**Comment by hadipournigjeh** [comment]
done number 100 till 199

**Comment by hadipournigjeh** [comment]
https://cyf-hadi-full-stack.netlify.app/

File: client/src/App.js [hunk 1]
```diff
@@ -1,11 +1,158 @@
...
+      alert("Please enter a valid title.");
+      return;
+    }
+
+    if (!newVideoUrl.includes("youtube.com")) {
+      alert("Please enter a valid YouTube URL.");
+      return;
+    }
+
+    // const newVideoId = Date.now().toString(); // Generate unique ID using current timestamp
```
Comment (sampennington) [inline client/src/App.js:66]: try to clean up code you no longer need! 😄

File: client/src/Video.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      updateRating(info.id, currentRating + 1);
+    }
+  }
+
+  function handleDislike() {
+    if (info.id && currentRating > 0) {
+      setCurrentRating(currentRating - 1);
+      updateRating(info.id, currentRating - 1);
+    }
+  }
```
Comment (sampennington) [inline client/src/Video.jsx:23]: I like the way you;'ve split this logic up 🎉

File: server/server.js [hunk 3]
```diff
@@ -1,15 +1,88 @@
...
+    const videoId = req.params.id;
+    const { rating } = req.body;
+    const query = "UPDATE videos SET rating=$1 WHERE id=$2";
+    const values = [rating, videoId];
+    await pool.query(query, values);
+    res.json({
+      message: "rating updated",
+    });
+  } catch (error) {
+    console.log(error);
```
Comment (sampennington) [inline server/server.js:85]: Any other status codes than 500 we should be returning here?

I think a 200 and a 400 could also be valid here, where might they go?

**Review by sampennington — COMMENTED** [review]
Nice Hadi, looks neat, well structured. Well done

File: client/src/App.js (hunk 1 above)
Comment (hadipournigjeh) [reply client/src/App.js:66]: Ok , thank you Sam , I'll delete it

File: client/src/Video.jsx (hunk 2 above)
Comment (hadipournigjeh) [reply client/src/Video.jsx:23]: Thanks a lot

File: server/server.js (hunk 3 above)
Comment (hadipournigjeh) [reply server/server.js:85]: I'll think about it how can I solve it

**Pull request closed without merging**